lib/buffer.js

```diff
@@ -892,8 +892,12 @@ function fill_(buf, val, start, end, encoding) {
 
   start = start >>> 0;
   end = end === undefined ? buf.length : end >>> 0;
+  const fillLength = bindingFill(buf, val, start, end, encoding);
 
-  return bindingFill(buf, val, start, end, encoding);
+  if (fillLength === -1)
+    throw new errors.Error('ERR_INVALID_ARG_VALUE', 'value', val);
```

Comment:
This should be a `TypeError` as far as I see it.

Reply:
+1... `TypeError` would be better here.